src/cache/colony/balances.ts

```diff
@@ -0,0 +1,109 @@
+import { BigNumber } from 'ethers';
+import {
+  DEFAULT_NETWORK_TOKEN,
+  ADDRESS_ZERO,
+  COLONY_TOTAL_BALANCE_DOMAIN_ID,
+} from '~constants';
+
+/*
+ * If token has Address Zero as it's address return local data representing
+ * the local chain token, otherwise just return the token
+ */
+const getTokenDataWithNative = (token) => {
+  if (token.id !== ADDRESS_ZERO) {
+    return token;
+  }
+  return {
+    ...token, // needed to add the correct data types
+    ...DEFAULT_NETWORK_TOKEN,
+  };
+};
+
+const BALANCE_TYPE_NAME = 'ColonyBalance';
+
+const balancesFieldCache = {
+  /*
+   * @NOTE Calculate total token balances across all domains (domain 0)
+   *
+   * This is based on all the token balances available in all the other domains,
+   * including root
+   */
+  balances: {
+    read: (domainBalanceRefs, { readField, cache }) => {
+      const colonyAddress = readField('id');
+      const { chainId: colonyChainId } = readField('meta');
+      const { items: colonyTokens = [] } = readField('tokens');
+      /*
+       * We're recreating the balances array in order to change the native chain
+       * token data to include the correct one from the one currently deployed
+       *
+       * @NOTE This is actually a wrong way to go about it, and once Multisig
+       * is upon us, we should be using the chain the colony is deployed to
+       * in order to dynamically contruct this (since multiple colonies
+       * can be deployed to multiple chains)
+       *
+       * @WARNING we have to get the full domain, and basically recreate the
+       * whole balance object (including the typename) since, if the alternative
+       * is true, and we pass the reference along, it will actually prevent
+       * us from returning the modified data, and will overwrite our returned
+       * results with the ones already available in the cache.
+       */
+      const allDomainBalances = domainBalanceRefs.items.map((balanceRef) => {
```

Comment:
I wonder if this also could be defined as a function that lives outside this main function, e.g. `const getAllDomainBalances = (domainRefs, readField, cache) => {...}`

Mainly from a readability perspective. Seeing four consecutive `return`s followed by an object is a little hard to follow.

Reply:
Don't care pro or con here, so If you insist, sure.